code/modules/projectiles/projectile/special/meteor.dm

```diff
@@ -6,17 +123,284 @@
 	damage_type = BRUTE
 	nodamage = TRUE
 	flag = BULLET
+	mouse_opacity = MOUSE_OPACITY_ICON
+	var/hits = 4
+	var/hitpwr = 2 //Level of ex_act to be called on hit.
+	var/dest
+	pass_flags = PASSTABLE | PASSGLASS | PASSGRILLE | PASSCLOSEDTURF | PASSMACHINE | PASSSTRUCTURE | PASSDOORS | PASSMOB | PASSGIRDER
+	var/heavy = FALSE
+	var/meteorsound = 'sound/effects/meteorimpact.ogg'
+	var/z_original
+	var/threat = 0 // used for determining which meteors are most interesting
+	var/lifetime = DEFAULT_METEOR_LIFETIME
+	var/timerid = null
+	var/list/meteordrop = list(/obj/item/stack/ore/iron)
+	var/dropamt = 2
+	range = 100
+
+/obj/projectile/meteor/ex_act()
+	return
+
+/obj/projectile/meteor/Initialize(mapload)
+	. = ..()
+	z_original = z
+	GLOB.meteor_list += src
+	SSaugury.register_doom(src, threat)
+	SpinAnimation()
+	timerid = QDEL_IN(src, lifetime)
+
+
+/obj/projectile/meteor/proc/make_debris()
+	for(var/throws = dropamt, throws > 0, throws--)
+		var/thing_to_spawn = pick(meteordrop)
+		new thing_to_spawn(get_turf(src))
+
+/obj/projectile/meteor/proc/meteor_effect()
+	if(heavy)
+		var/sound/meteor_sound = sound(meteorsound)
+		var/random_frequency = get_rand_frequency()
+
+		for(var/mob/M in GLOB.player_list)
+			if((M.orbiting) && (SSaugury.watchers[M]))
+				continue
+			var/turf/T = get_turf(M)
+			if(!T || T.z != src.z)
+				continue
+			var/dist = get_dist(M.loc, src.loc)
+			shake_camera(M, dist > 20 ? 2 : 4, dist > 20 ? 1 : 3)
+			M.playsound_local(src.loc, null, 50, 1, random_frequency, 10, S = meteor_sound)
+
+/obj/projectile/meteor/proc/ram_turf(turf/T)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names